sbi/inference/snpe/snpe_base.py

```diff
@@ -365,6 +371,9 @@ def _train(
         # to the untrained copy.
         if self._retrain_from_scratch_each_round and round_ > 0:
             self._posterior = deepcopy(self._untrained_neural_posterior)
+            optimizer = optim.Adam(
+                list(self._posterior.net.parameters()), lr=learning_rate,
```

Comment:
I am surprised that this list-wrapping is needeed...

Besides, I see a chance to write a method  `_handle_retrain...(self, round_, learning_rate)` at NeuralInference and reuse it across children?

Reply:
I think you're right, this is probably not needed. I'll check.

It's not so simple because for SNRE the summary net is not part of `posterior.net`, but a separate property.

Reply:
I think we need to talk about this, there are multiple issues at once. Open a chat when it suits you.